src/functional/ffront/foast_passes/closure_var_type_deduction.py

```diff
@@ -0,0 +1,59 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2022, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either vervisit_Constantsion 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from dataclasses import dataclass
+from typing import Any
+
+import functional.ffront.field_operator_ast as foast
+from eve import NodeTranslator, traits
+from functional.ffront.symbol_makers import make_symbol_type_from_value
+
+
+@dataclass(frozen=True)
+class ClosureVarTypeDeduction(NodeTranslator, traits.VisitorWithSymbolTableTrait):
+    """
+    Deduce the type of `Symbol` nodes that introduce closure variables.
+
+    The types are deduced by looking at the values inside the list of
+    closure variables. The types are inferred from the Python value.
+
+    For the general type deduction pass, all Symbol nodes must already be typed,
+    this pass must run before.
+    """
```

Comment:
```suggestion
    Deduce the type of all closure variable declarations.
```
Too technical, you can add the details afterwards if you want.
```suggestion
    Deduce the type of all closure variable declarations.
    
    After this pass all closure var declarations, i.e. the `Symbol`s contained in 
    `foast.FunctionDefinition.closure_vars`, have the type as specified by their
    actual values (given by `closure_vars` to `apply`).
```

Reply:
Why do you think it's too technical? It's still concise in one line, says exactly what it does, and I think speaking in terms of the AST is appropriate when you are manipulating the AST.